tensorflow/contrib/ignite/kernels/ignite_dataset_ops.cc

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/contrib/ignite/kernels/ignite_holder.h"
+
+using namespace ignite;
+using namespace cache;
+using namespace query;
+using namespace binary;
+
+namespace tensorflow {
+
+    class IgniteDatasetOp : public DatasetOpKernel {
+    public:
+        using DatasetOpKernel::DatasetOpKernel;
+
+        void MakeDataset(OpKernelContext *ctx, DatasetBase **output) override {
+            std::string cache = "";
+            OP_REQUIRES_OK(ctx, ParseScalarArgument<std::string>(ctx, "cache", &cache));
+            *output = new Dataset(ctx, cache);
+        }
+
+    private:
+        class Dataset : public GraphDatasetBase {
+        private:
+            Ignite grid;
+            const std::string cacheName;
+        public:
+            Dataset(OpKernelContext *ctx, const string &cache)
+                    : GraphDatasetBase(ctx), cacheName(cache) {
+                grid = IgniteHolder::Instance().getIgnite();
+            }
+
+            std::unique_ptr<IteratorBase> MakeIteratorInternal(
+                    const string &prefix) const override {
+                return std::unique_ptr<IteratorBase>(
+                        new Iterator({this, strings::StrCat(prefix, "::Ignite")}));
+            }
+
+            const DataTypeVector &output_dtypes() const override {
+                static DataTypeVector *dtypes = new DataTypeVector({DT_STRING});
+                return *dtypes;
+            }
+
+            const std::vector<PartialTensorShape> &output_shapes() const override {
+                static std::vector<PartialTensorShape> *shapes =
+                        new std::vector<PartialTensorShape>({{}});
+                return *shapes;
+            }
+
+            string DebugString() const override { return "IgniteDatasetOp::Dataset"; }
+
+        protected:
+            Status AsGraphDefInternal(DatasetGraphDefBuilder *b,
+                                      Node **output) const override {
+                return Status::OK();
```

Comment:
This should return `errors::Unimplemented()`.

Reply:
Fixed.